### ISSUE TITLE
Language server doesn't respond to initialize request properly

### ISSUE THREAD

**Issue by rozaliev** [tailwindlabs/tailwindcss-intellisense#802]
I'm trying to make language-server work with Helix. But it looks like lsp doesn't respond to initialize request properly before sending its own requests. According to the [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#lifeCycleMessages): 

> Until the server has responded to the initialize request with an [InitializeResult](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initializeResult), the client must not send any additional requests or notifications to the server. In addition the server **is not allowed to send any requests** or notifications to the client until it has responded with an [InitializeResult](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initializeResult), with the exception that during the initialize request the server is allowed to send the notifications window/showMessage, window/logMessage and [telemetry/event](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#telemetry_event) as well as the window/showMessageRequest request to the client.

Language server:  @tailwindcss/language-server@0.0.13
`tailwind-language-server --stdio`

Expected:
```
-> {"jsonrpc":"2.0","method":"initialize","params":{ ... },"id":0}
<- {"jsonrpc":"2.0","id":0,"result": { ... }}
```

Received:
```
-> {"jsonrpc":"2.0","method":"initialize","params":{ ... },"id":0}
<- {"jsonrpc":"2.0","id":0,"method":"workspace/configuration","params":{"items":[{"section":"editor"}]}}
<- {"jsonrpc":"2.0","id":1,"method":"workspace/configuration","params":{"items":[{"section":"tailwindCSS"}]}}
```

I would assume that this line is the reason https://github.com/tailwindlabs/tailwindcss-intellisense/blob/1438300d98f23bd0c80a65759e88599d925d9783/packages/tailwindcss-language-server/src/server.ts#L1655

And this these are the lines that send requests **before** the initialize response. https://github.com/tailwindlabs/tailwindcss-intellisense/blob/1438300d98f23bd0c80a65759e88599d925d9783/packages/tailwindcss-language-server/src/server.ts#L247

**Comment by atgote** [comment]
I can confirm absolutely the same behavior. I'm using vim 9.
However the same server binary works well under coc.nvim plugin but is not working under https://github.com/yegappan/lsp.
Looks like something is not good with plugin (read 'client') caps...

**Comment by atgote** [comment]
@rozaliev You may try to use my fork for testing (will require you to build packages `lerna run build`, set execution permission on the resulting `packages/tailwindcss-landuage-server/dist/.../bin/tailwindcss-landuage-server` and using path to this executable in your helix lsp config)

**Comment by rozaliev** [comment]
@atgote thanks, your fork seems to work with Helix, at least for the initialization part and simple HTML editing.

But with custom regex, only completion works and hover doesn't. Here are the weird things I found in the logs:

```
2023-06-20T11:02:54.772 helix_lsp::transport [INFO] tailwind-hack -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-8","utf-32","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":true,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"inlayHint":{"dynamicRegistration":false},"publishDiagnostics":{"versionSupport":true},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":true},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"executeCommand":{"dynamicRegistration":false},"inlayHint":{"refreshSupport":false},"symbol":{"dynamicRegistration":false},"workspaceEdit":{"documentChanges":true,"failureHandling":"abort","normalizesLineEndings":false,"resourceOperations":["create","rename","delete"]},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"23.05"},"initializationOptions":{"tailwindCSS":{"experimental":{"classRegex":["class: \"(.*)\""]},"includedLanguages":{"rust":"html"}}},"processId":60856,"rootPath":"/mypath/diox_1","rootUri":"file:///mypath/diox_1","workspaceFolders":[{"name":"diox_1","uri":"file:///mypath/diox_1"}]},"id":0}


2023-06-20T11:02:55.493 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, MethodCall(MethodCall { jsonrpc: Some(V2), method: "client/registerCapability", params: Map({"registrations": Array [Object {"id": String("f96c2630-3ce7-4ec3-8b5d-d51b667f4dee"), "method": String("textDocument/hover"), "registerOptions": Object {"documentSelector": Null}}, Object {"id": String("c07e4872-a44f-4098-90d1-0597b761c22b"), "method": String("textDocument/documentColor"), "registerOptions": Object {"documentSelector": Null}}, Object {"id": String("6f3342de-6b09-4b1e-a069-51788dd061e6"), "method": String("textDocument/codeAction"), "registerOptions": Object {"documentSelector": Null}}, Object {"id": String("349075c1-391a-4393-847a-2ab7b1616e2e"), "method": String("textDocument/documentLink"), "registerOptions": Object {"documentSelector": Null}}, Object {"id": String("8db816d3-1c8c-41a6-a95e-75622fb8d660"), "method": String("textDocument/completion"), "registerOptions": Object {"documentSelector": Null, "resolveProvider": Bool(true), "triggerCharacters": Array [String("\""), String("'"), String("`"), String(" "), String("."), String("("), String("["), String("!"), String("/"), String(":")]}}]}), id: Num(4) })))

2023-06-20T11:02:55.493 helix_term::application [WARN] Ignoring a client/registerCapability request because dynamic capability registration is not enabled. Please report this upstream to the language server
```
I'm not sure if this is directly related to hover, because it works for HTML and shows similar errors. But it definitely seems like a bug, `dynamicRegistration` is clearly set to false by the client and is ignored.

**Comment by rozaliev** [comment]
I think I figured out how to make hover work with custom regex, but I'm not sure how to proceed since I'm not really familiar with Tailwind codebase and there are no tests.


I assume that somewhere around here should be code that handles `includeLanguages` and `classRegex` 

https://github.com/tailwindlabs/tailwindcss-intellisense/blob/1438300d98f23bd0c80a65759e88599d925d9783/packages/tailwindcss-language-service/src/util/find.ts#L446


I've added this line to test things out:
```ts
else {
    classNames = await findClassNamesInRange(state, doc, searchRange, 'html', true)
}
```
Notice the last argument, which sets `includeCustom` to true and enables `classRegex`

**Comment by bradlc** [comment]
This should be fixed by #803 but please open a new issue if you're still running into problems 👍 

We now automatically publish `insiders` releases for `@tailwindcss/language-server`, so until we publish a new `latest` release that's the one to try out (`npm install @tailwindcss/language-server@insiders`)
